iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/IcebergTableUtil.java

```diff
@@ -484,6 +484,10 @@ public static List<FieldSchema> getPartitionKeys(Table table, int specId) {
             String.format("Transform: %s", partField.transform().toString()))).collect(Collectors.toList());
   }
 
+  public static List<FieldSchema> getAllPartitionKeys(Table table) {
+    return  table.specs().keySet().stream().flatMap(id -> getPartitionKeys(table, id).stream())
```

Comment:
why do we need double loops? can we push `latestSpecOnly` to getPartitionKeys and only apply the filter if latestSpecOnly=true?

Reply:
Done.